backend/apps/progress/signals.py

```
@@ -132,3 +132,22 @@ def on_exercise_attempt(sender, instance, created, **kwargs):
+            from django.db import transaction
+            from django_q.tasks import async_task
+            
+            transaction.on_commit(
+                lambda: async_task(
+                    "apps.progress.tasks.update_leaderboard_task", 
+                    instance.user.id,
+                    instance.user.username,
+                    instance.xp_delta
+                )
+            )
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)